src/qz/common/TrayManager.java

```diff
@@ -402,7 +395,7 @@ public boolean showGatewayDialog(final Certificate cert, final String prompt) {
                     log.info("Denied {} to {}", cert.getCommonName(), prompt);
                     if (gatewayDialog.isPersistent()) {
                         if (Certificate.UNKNOWN.equals(cert)) {
-                            anonymousItem.doClick(); // if always block anonymous requests -> flag menu item
+                            anonymousItem.setChecked(true); // if always block anonymous requests -> flag menu item
```

Comment:
is this correct? I read it as, it toggles only when it is false, so we could just set it to true instead.